### PR TITLE
Add linear structure

### PR DESCRIPTION
**Pull request by tmigot**
- check the jacobian
- benchmark on examples with linear constraints

### CODE REVIEW

**Comment by tmigot** [comment]
@JSOBot runtests

**Comment by tmigot** [comment]
@JSOBot runtests

File: src/nlp.jl [hunk 1]
```diff
@@ -51,6 +51,12 @@ function KnitroSolver(
     ucon = copy(nlp.meta.ucon)
     ucon[uconinf] .= KNITRO.KN_INFINITY
   end
+  if nlp.meta.nlin > 0
+    jac_lin = jac(nlp, nlp.meta.x0)
```
Comment (dpo) [inline src/nlp.jl:55]: Shouldn't we use `jac_lin_structure` and `jac_lin_coord` here?

File: src/nlp.jl [hunk 2]
```diff
@@ -81,10 +87,12 @@ function KnitroSolver(
 
     if evalRequestCode == KNITRO.KN_RC_EVALFC
       evalResult.obj[1] = obj(nlp, x)
-      m > 0 && cons!(nlp, x, evalResult.c)
+      m > 0 &&cons!(nlp, x, evalResult.c)
```
Comment (dpo) [inline src/nlp.jl:90]: `cons_lin!` ?

**Comment by tmigot** [comment]
Superseded by #88

**Pull request closed without merging**